### DISCUSSION TITLE
Improve workflow / focus issues

### DISCUSSION THREAD

**Discussion by notDavid** [keepassxreboot/keepassxc#8481] [Q&A]
Hi there,

So i've noticed a few issues with the Firefox browser extension which hopefully can be improved, please let me know if i should create separate issues for these;

1.  Don't require to click "Unlock Database" for TouchID when "Fill" button is clicked in Firefox
    - this should automatically popup the TouchID dialog
2.  Focus does not return to Firefox (fullscreen) after authenticating with TouchID
    - Focus should automatically go back to Firefox
    - The KeepassXC window was previously minimized/hidden to the Menubar. Please return back to the hidden/minimized KeepassXC window state after authenticating.
3.  Fill button does not work + "redetect login fields" does not work
    - requires a page refresh
4. Focus does not change to KeepassXC when asking whether to allow access to specific entry
   - Also note that focus now DOES return back to Firefox successfully (see `#2`)
   - Please return back to the hidden/minimized KeepassXC window state after returning focus to Firefox.

(this is a new, default Firefox profile.)

Screen recording:

https://user-images.githubusercontent.com/1530401/190997321-e6efa468-f33b-49a2-929c-4f7330581f5a.mp4

**Comment by droidmonkey** [comment]
A large part of this is because you are running everything (it appears) as a fullscreen app (desktop?) in macOS... that breaks a few workflows currently.

The first bullet will only be fixed if there is one database open, its on the list already.

The third bullet is an extension problem, I believe that has been fixed in the latest develop branch https://github.com/keepassxreboot/keepassxc-browser

**Reply by notDavid** [reply]
Thank you for your reply;

> because you are running everything (it appears) as a fullscreen app (desktop?)

Not sure what you mean by everything, but only Firefox is running in Fullscreen.
Also note that in point 4 focus does return back to Firefox like it should.

**Reply by droidmonkey** [reply]
Yes noted, will take a look at the unlock workflow, might be missing an explicit "fall to the background" when done from the Browser interface.
